index.js

```diff
@@ -122,10 +122,22 @@ const plugin = fp(async function (app, opts) {
     schema = gateway.schema
     entityResolversFactory = gateway.entityResolversFactory
 
+    if (gateway.pollingInterval !== undefined) {
+      setInterval(async () => {
+        try {
+          fastifyGraphQl.schema = await gateway.refresh()
+        } catch (err) {
+          console.error(err)
+        }
+      }, gateway.pollingInterval)
```

Comment:
this interval must be cancelled during the `app.onClose()` block down below.

Reply:
I have added a condition into app.onClose

```js
if (gatewayInterval !== null) {
    clearInterval(gatewayInterval)
}
```